### PR TITLE
DOC: Fix broken links in VARResults

### PR DESCRIPTION
**Pull request by bashtage**
Move links out of first line of docstring to avoid broken links

### CODE REVIEW

File: statsmodels/tsa/vector_ar/var_model.py [hunk 1]
```diff
@@ -1579,10 +1579,7 @@ def reorder(self, order):
 
     def test_causality(self, caused, causing=None, kind='f', signif=0.05):
         """
-        Test for Granger-causality as described in chapter 7.6.3 of [1]_.
-        Test H0: "`causing` does not Granger-cause the remaining variables of
-        the system" against  H1: "`causing` is Granger-causal for the
-        remaining variables".
+        Test Granger causality
```
[the remark this answers is not in the document]
Comment (josef-pkt) [reply statsmodels/tsa/vector_ar/var_model.py]: see https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt
`3. Extended Summary`

The main important information should go after the 1-line summary.
We do, or better, should do this in most places when the purpose is not "obvious" from the 1-line summary.

[the remark this answers is not in the document]
Comment (bashtage) [reply statsmodels/tsa/vector_ar/var_model.py]: I don't think telling an end user what a test's null is would qualify as clarifying functionality.

**Comment by jbrockmendel** [comment]
FWIW, experience has taught me to trust @bashtage to be [Technically Correct](https://www.youtube.com/watch?v=hou0lU8WMgo) when it comes to docstring formatting.  The suggested change to test_inst_causality is definitely an improvement.

BTW there's a testgrangercausality in one of the many stats/tsastats/stattsa/toolstats modules that I'm pretty sure is never used outside of the test suite.  Can it be re-used here?

**Comment by josef-pkt** [comment]
@jbrockmendel What do you mean "is never used outside of the test suite"?
It's a standalone hypothesis test.

The difference is that `grangercausalitytests` is a univariate test using a univariate regression (one series granger causes another), while the one in VAR/VECM is for a multivariate system that allows more choices of causing and caused variables.

File: statsmodels/tsa/vector_ar/var_model.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (josef-pkt) [reply statsmodels/tsa/vector_ar/var_model.py]: it is/was to me #305
in several tests like adfuller versus kpss it is not clear what the null hypothesis is, unless you remember your econometrics, which I didn't or didn't pay enough attention to

**Comment by josef-pkt** [comment]
merged

@bashtage  Thanks

I leave the H0, H1 for another day,
e.g. I got tired of figuring out what Null and Alternatives are and started to add full extended summaries like https://github.com/statsmodels/statsmodels/pull/2723/files#diff-8d9af3a01c72248b06370f02ea4a88b7R34
(half of the time I don't remember if a two sample test has one-sided alternative m1 > m2 or m1 < m2, even though I wrote a large number of those.)

**Comment by jbrockmendel** [comment]
> What do you mean "is never used outside of the test suite"?  It's a standalone hypothesis test.

In my personal vernacular it means not used elsewhere in statsmodels.

**Pull request merged**